src/types.ts

```diff
@@ -222,5 +222,11 @@ export type ApiError = {
 };
 
 export const isApiError = (err: unknown): err is ApiError => {
-  return typeof err === 'object' && typeof err['json'] === 'object';
+  return (err as ApiError).json !== undefined;
```

Comment:
Yes, and that is correct. `isApiError` returns true if the error has `json` object property. My changes retain the same functionality as original code but it makes it compliant with typescript rules.

Reply:
can this work?
`typeof err === 'object' && err.json` 
err is unknown, it might not be an object and if it's a string for example, it'll cause the error handling to crash :(

Reply:
`typeof err === 'object' && err.json` causes typescript error saying `json` does not exist on type object or something along these lines.

Why would it cause the error handling to crash? This function is simple typeguard which checks if `json` property exists on `err` object. It fails in other cases. The function which correctly derives an error message from the unknown type Error is `getErrorMessage` in utils.ts

Reply:
Right, it'll fail only if err is null or undefined